File: src/migration/00-init-index.ts
import Database, { mongoClient } from "../server/db";

(async () => {
    await mongoClient.connect();

    const db = new Database();

    db.user.createIndex({email: 1}, {unique: true});

    db.template.createIndex({userId: 1, name: 1}, {unique: true});
    db.template.createIndex({userId: 2, front: 2}, {unique: true});

    db.templateData.createIndex({templateId: 1, entry: 1}, {unique: true});

    db.deck.createIndex({userId: 1, name: 1}, {unique: true});

    db.card.createIndex({userId: 1, front: 1}, {unique: true});

    mongoClient.close();
})().catch((e) => console.error(e));

File: py/migration/00_init_index.py
from py.migration.conn import collection
import pymongo

if __name__ == "__main__":
    collection.user.create_index([("email", pymongo.TEXT)], unique=True)
    collection.deck.create_index([("userId", 1), ("name", 1)], unique=True)
    collection.card.create_index([("userId", 1), ("front", 1)], unique=True)
    collection.media.create_index([("sourceId", 1), ("h", 1)], unique=True)
